packages/language-service/ivy/definitions.ts

```diff
@@ -6,47 +6,35 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {AST, TmplAstNode} from '@angular/compiler';
+import {AST, flatten, TmplAstNode, TmplAstTextAttribute} from '@angular/compiler';
 import {NgCompiler} from '@angular/compiler-cli/src/ngtsc/core';
 import {ShimLocation, Symbol, SymbolKind} from '@angular/compiler-cli/src/ngtsc/typecheck/api';
 import * as ts from 'typescript';
 
 import {findNodeAtPosition} from './hybrid_visitor';
-import {getTemplateInfoAtPosition, getTextSpanOfNode, isDollarEvent, toTextSpan} from './utils';
+import {getDirectiveMatches, getDirectiveMatchesForAttribute, getTemplateInfoAtPosition, getTextSpanOfNode, isDollarEvent, toTextSpan} from './utils';
 
 export class DefinitionBuilder {
   constructor(private readonly tsLS: ts.LanguageService, private readonly compiler: NgCompiler) {}
 
-  // TODO(atscott): getTypeDefinitionAtPosition
-
   getDefinitionAndBoundSpan(fileName: string, position: number): ts.DefinitionInfoAndBoundSpan
       |undefined {
-    const templateInfo = getTemplateInfoAtPosition(fileName, position, this.compiler);
-    if (templateInfo === undefined) {
-      return undefined;
-    }
-    const {template, component} = templateInfo;
-
-    const node = findNodeAtPosition(template, position);
+    const {symbol, node} = this.getSymbolAndNodeAtPosition(fileName, position);
     // The `$event` of event handlers would point to the $event parameter in the shim file, as in
     // `_outputHelper(_t3["x"]).subscribe(function ($event): any { $event }) ;`
     // If we wanted to return something for this, it would be more appropriate for something like
     // `getTypeDefinition`.
-    if (node === undefined || isDollarEvent(node)) {
-      return undefined;
-    }
-
-    const symbol = this.compiler.getTemplateTypeChecker().getSymbolOfNode(node, component);
-    if (symbol === null) {
+    if (symbol === undefined || node === undefined || isDollarEvent(node)) {
       return undefined;
     }
-
     const definitions = this.getDefinitionsForSymbol(symbol, node);
     return {definitions, textSpan: getTextSpanOfNode(node)};
   }
 
   private getDefinitionsForSymbol(symbol: Symbol, node: TmplAstNode|AST):
       readonly ts.DefinitionInfo[]|undefined {
+    const getDefinitions = (...symbols: {shimLocation: ShimLocation}[]) =>
```

Comment:
nit (google style guide)

```suggestion
    const getDefinitions = (...symbols: Array<{shimLocation: ShimLocation}>) =>
```

Reply:
Is there a reason for adding this change here other than as a fly-by refactor?